### PR TITLE
 32-function-for-accessing-the-users-current-language 

### PR DESCRIPTION
**Pull request by durveshps**
### SUMMARY
Implements necessary changes for #32

*Reviewers:* @numere-org/maintainers

### IMPLEMENTATION
* Implementation: (*Describe, what you've changed*) 
* Implementation test: (*Describe the type of test, which you performed, and if it was successful*)

### DOCUMENTATION
* [ ] ChangesLog updated
* [x] Code changes commented
* **Documentation articles:**
    * [x] corresponding documentation articles updated
    * [ ] new documentation articles created
    * [ ] not needed
* **Language files:**
    * [x] corresponding language files updated
    * [ ] not needed

----------------

### TESTS BY REVIEWERS
* [x] Added to the automatic SW tests
* [ ] Tested manually

### CODE REVIEW

File: kernel/core/strings/stringfunctions.hpp [hunk 1]
```diff
@@ -29,6 +29,8 @@
 // define the "End of transmission block" as string separator
 #define NEWSTRING (char)23
 
+extern Language _guilang;
```
Comment (numeredev) [inline kernel/core/strings/stringfunctions.hpp:32]:
```suggestion
extern Language _lang;

```
Works with both, but I'd like to have both instances separated.

**Review by numeredev — CHANGES_REQUESTED** [review]
Please use `_lang` in the kernel component and `_guilang` in the GUI component.

**Comment by numeredev** [comment]
Needs the addition of `getuilang()` at the other locations in code (about two).

**Comment by durveshps** [comment]
A side question, I was trying to test this, but it threw error that getuilang function does not exist. Is it also becuase I used _guilang and not _lang?

**Comment by numeredev** [comment]
> A side question, I was trying to test this, but it threw error that getuilang function does not exist. Is it also becuase I used _guilang and not _lang?

Nope, I think that's because the parser does not yet know, how to find the function. For testing, try to enter `getuilang() + ""`. For production, you've to search for occurences of `getversioninfo(` in code and add `getuilang(` to those conditions.

**Review by numeredev — COMMENTED** [review]
Changes accepted. Will be merged after tests have been implemented.

**Comment by numeredev** [comment]
> Changes accepted. Will be merged after tests have been implemented.

@durveshps You're missing the entry in the ChangesLog. Please add that.

**Pull request merged**